ledger/src/transaction_address_lookup_table_scanner.rs

```diff
@@ -0,0 +1,56 @@
+use {
+    bincode::deserialize,
+    lazy_static::lazy_static,
+    solana_sdk::{
+        address_lookup_table::{self, instruction::ProgramInstruction},
+        pubkey::Pubkey,
+        sdk_ids::SDK_IDS,
+        transaction::SanitizedVersionedTransaction,
+    },
+    std::collections::HashSet,
+};
+
+lazy_static! {
+    static ref SDK_IDS_SET: HashSet<Pubkey> = SDK_IDS.iter().cloned().collect();
+}
+
+#[derive(Default)]
+pub struct ScannedLookupTableExtensions {
+    pub possibly_incomplete: bool,
+    pub accounts: Vec<Pubkey>, // empty if no extensions found
+}
+
+pub fn scan_transaction(
+    transaction: &SanitizedVersionedTransaction,
+) -> ScannedLookupTableExtensions {
+    // If the ALT program is not present in the account keys, it was not used
+    if !transaction
+        .get_message()
+        .message
+        .static_account_keys()
+        .iter()
+        .any(address_lookup_table::program::check_id)
+    {
+        return ScannedLookupTableExtensions::default();
+    }
```

Comment:
i think we still need to return `possibly_incomplete: true` for the possibility of